block/manager.go

```diff
@@ -189,19 +207,53 @@ func (m *Manager) AggregationLoop(ctx context.Context) {
 		time.Sleep(delay)
 	}
 
+	//var timer *time.Timer
 	timer := time.NewTimer(0)
 
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case <-timer.C:
-			start := time.Now()
-			err := m.publishBlock(ctx)
-			if err != nil {
-				m.logger.Error("error while publishing block", "error", err)
+	if !lazy {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-timer.C:
+				start := time.Now()
+				err := m.publishBlock(ctx)
+				if err != nil {
+					m.logger.Error("error while publishing block", "error", err)
+				}
+				timer.Reset(m.getRemainingSleep(start))
+			}
+		}
+	} else {
+		<-timer.C
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			// the buildBlock channel is signalled when Txns become available
+			// in the mempool, or after transactions remain in the mempool after
+			// building a block.
+			case <-m.buildBlock:
```

Comment:
i think your earlier logic was better, where you start the timer on `m.txsAvailable` and build the block once the `1s` timer expires. i don't think we need to complicate the logic here to capture any pending txns which might have gotten while the lazy sequencer is building the block. Since it is already a lazy sequencer, i think it is okay to let those txns wait for future `txsAvailable` notification. what do you think?

Reply:
Yes that would be much simpler